### PR TITLE
Improved package publishing detection and handling

### PR DESCRIPTION
**Pull request by solarce**
- Detects if a package already exists on package cloud
  - If the package exists and is a dev build, it removes it
    and publishes it again
  - If the package exists and is a release build, it prompts
    for manual conflict resolution and the build fails
- This all works on a per package (distro + distro release + type) basis

### CODE REVIEW

File: utils/pkg/pkg_run [hunk 1]
```diff
@@ -63,12 +72,43 @@ docker_run () {
...
+      export PC_REPO="worker"
+    fi
+
+    if [[ $pkgtype == "rpm" ]]; then
+      PC_VERSION=$(echo $VERSION|sed 's/\-/\_/')
+    else
+      PC_VERSION=$VERSION
+    fi
+
+    export LATEST_PUBLISHED_FILENAME=$(curl -s https://$(echo $PACKAGECLOUD_TOKEN):@packagecloud.io/api/v1/repos/$PC_USER/$PC_REPO/package/$pkgtype/$package_cloud_os/$platform_release/travis-worker/$pkgarch/$PC_VERSION/1.json|jq .'filename'|tr -d '"')
```
Comment (meatballhat) [inline utils/pkg/pkg_run]: I'm curious: why the `$(echo $PACKAGECLOUD_TOKEN)` as opposed to using `$PACKAGECLOUD_TOKEN` directly?

File: utils/pkg/pkg_run [hunk 2]
```diff
@@ -63,12 +72,43 @@ docker_run () {
...
+    if [[ $pkgtype == "rpm" ]]; then
+      export LATEST_PUBLISHED_VERSION=$(echo $LATEST_PUBLISHED_FILENAME|cut -d'-' -f 3|sed 's/_/-/')
+    else
+      export LATEST_PUBLISHED_VERSION=$(echo $LATEST_PUBLISHED_FILENAME|cut -d'_' -f 2)
+    fi
+
+    if [[ $VERSION == $LATEST_PUBLISHED_VERSION ]]; then
+      if [[ $VERSION =~ dev ]]; then
+        package_cloud yank travisci/$PC_REPO/$package_cloud_os/$platform_release $LATEST_PUBLISHED_FILENAME
```
Comment (meatballhat) [inline utils/pkg/pkg_run]: :clap:

**Comment by meatballhat** [comment]
:sheep: :it: LGTM

File: utils/pkg/pkg_run (hunk 2 above)
Comment (solarce) [inline utils/pkg/pkg_run]: The joys of differences in package managers :sob:

File: utils/pkg/pkg_run (hunk 1 above)
Comment (solarce) [inline utils/pkg/pkg_run]: I don't know why I have `echo` in there :smiley: but clearly it must be more better

**Comment by solarce** [comment]
Just wanted to note it took me a lot longer than I expected to sort out this logic because you have to delete on a per package file basis and the API docs had a couple errors that I was able to help the pcloud folks fix.

I am hoping to roll this (and some other improvements) into the packagecloud.io addon sometime in the next couple of weeks

**Pull request merged**